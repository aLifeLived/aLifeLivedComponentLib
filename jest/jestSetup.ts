/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
