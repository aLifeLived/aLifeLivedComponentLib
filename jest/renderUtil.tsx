import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
