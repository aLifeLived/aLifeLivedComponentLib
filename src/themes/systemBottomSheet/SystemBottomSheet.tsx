import React from 'react';
import { Box } from '../box/Box';
import { Button } from '../../components/button/Button';

type BottomSheetType = {
  isVisible: boolean;
  onClose: () => void;
  children: React.ReactNode;
};

export const SystemBottomSheet: React.FC<BottomSheetType> = ({
  isVisible,
  onClose,
  children,
}) => {
  return isVisible ? (
    <Box
      position="absolute"
      right={0}
      left={0}
      bottom={0}
      top={0}
      justifyContent="flex-end"
      backgroundColor="backgroundVisible"
      zIndex="foregroundTop"
    >
      <Box
        justifyContent="flex-end"
        flex={1}
        alignContent="center"
        alignItems="center"
        paddingBottom="s"
        borderTopLeftRadius="large"
        borderTopRightRadius="large"
        paddingTop="l"
      >
        {children}
        <Button
          title="Close"
          onPress={() => onClose()}
          backgroundColor="backgroundContrastXLow"
          color="textContrastHigh"
          marginTop="l"
          marginBottom="l"
        />
      </Box>
    </Box>
  ) : null;
};
