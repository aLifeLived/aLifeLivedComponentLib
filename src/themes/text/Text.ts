import { createText, ResponsiveValue } from '@shopify/restyle';
import { ColorThemeTokens } from '../palette';
import { Theme } from '../theme';
import { ValueOf } from '../../ts/utils';

const FontFamilies = {
  sfUiTextBold: 'SFUIText-Bold',
  sfUiTextRegular: 'SFUIText-Regular',
  sfUiTextSemiBold: 'SFUIText-Semibold',
  sfUiDisplayRegular: 'SFUIDisplay-Regular',
};

type TextVariantToken =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'intro'
  | 'body'
  | 'bodySmall'
  | 'bodyXSmall';

type TextVariant = {
  fontFamily: ValueOf<typeof FontFamilies>;
  fontSize: number | ResponsiveValue<number, Theme>;
  lineHeight?: number | string;
  color: ColorThemeTokens;
};

export const textVariants: Record<TextVariantToken, TextVariant> = {
  h1: {
    fontSize: 52,
    lineHeight: 52,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h2: {
    fontSize: 32,
    lineHeight: 40,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h3: {
    fontSize: 24,
    lineHeight: 28.8,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h4: {
    fontSize: 20,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextRegular,
  },
  intro: {
    fontSize: 16,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextSemiBold,
  },
  body: {
    fontSize: 18,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodySmall: {
    fontSize: 16,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodyXSmall: {
    fontSize: 14,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
};

export const Text = createText<Theme>();
