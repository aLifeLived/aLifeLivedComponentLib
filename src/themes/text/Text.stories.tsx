import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { Text } from './Text';
import { CenterView } from '../../components/centerView/CenterView';

storiesOf('Text', module)
  .addDecorator(Story => <CenterView children={Story()} />)
  .add('Text', () => (
    <>
      <Text variant="h1">H1 Heading</Text>
      <Text variant="h2">H2 Heading</Text>
      <Text variant="h3">H3 Heading</Text>
      <Text variant="h4">H4 Heading</Text>
      <Text variant="body">Body</Text>
    </>
  ));
