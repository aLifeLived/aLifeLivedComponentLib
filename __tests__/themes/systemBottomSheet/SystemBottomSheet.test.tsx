import React from 'react';
import { SystemBottomSheet } from '../../../src/themes/systemBottomSheet/SystemBottomSheet';
import { fireEvent, render } from '../../../jest/renderUtil';
import { Text } from '../../../src/themes/text/Text';

const mockOnClose = jest.fn();

describe('themes >> systemBottomSheet', () => {
  beforeEach(() => mockOnClose.mockClear());
  describe('Should render systemBottomSheet box with no children', () => {
    test('Render snapshot correctly', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find close button', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );
      expect(getByText('CLOSE')).not.toBeNull();
    });

    test('Should fire onClose()', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );

      fireEvent.press(getByText('CLOSE'));
      expect(mockOnClose).toBeCalledTimes(1);
    });
  });

  describe('Should NOT render systemBottomSheet', () => {
    test('Should not render bottom sheet', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={false} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Should render systemBottomSheet with children', () => {
    test('Should render bottom sheet with text', () => {
      const { toJSON } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          <Text>Hi there</Text>
        </SystemBottomSheet>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find text', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          <Text>Hi there</Text>
        </SystemBottomSheet>
      );
      expect(getByText('Hi there')).not.toBeNull();
    });

    test('Should find close button', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );
      expect(getByText('CLOSE')).not.toBeNull();
    });

    test('Should fire onClose() with children', () => {
      const { getByText } = render(
        <SystemBottomSheet isVisible={true} onClose={mockOnClose}>
          {}
        </SystemBottomSheet>
      );

      fireEvent.press(getByText('CLOSE'));
      expect(mockOnClose).toBeCalledTimes(1);
    });
  });
});
