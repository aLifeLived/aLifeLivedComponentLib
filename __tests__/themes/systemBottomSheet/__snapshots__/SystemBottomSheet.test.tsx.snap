// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> systemBottomSheet Should NOT render systemBottomSheet Should not render bottom sheet 1`] = `null`;

exports[`themes >> systemBottomSheet Should render systemBottomSheet box with no children Render snapshot correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingTop": 24,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "marginBottom": 24,
          "marginTop": 24,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontFamily": "SFUIText-Semibold",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;

exports[`themes >> systemBottomSheet Should render systemBottomSheet with children Should render bottom sheet with text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingTop": 24,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
        ]
      }
    >
      Hi there
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "marginBottom": 24,
          "marginTop": 24,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontFamily": "SFUIText-Semibold",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;
